File: src/main/java/model/equipment/weapon/Axe.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class Axe extends Weapon {
    public Axe(double weight, int quality) {
        super(WeaponType.AXE, weight, quality);
        isOneHanded = true;
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.BLUNT, quality * 0.25);
        damage.put(DmgType.SLASH, quality * 0.75);
    }
}

File: src/main/java/model/equipment/weapon/Axe.kt
package model.equipment.weapon

import model.support.enums.Skill
import model.support.enums.Skill.AXE_SKILL
import model.support.enums.Skill.ONE_HANDED_WEAPON_SKILL
import model.support.enums.Specification.*
import model.support.enums.WeaponType.AXE

class Axe(weight: Double, quality: Int) : Weapon(AXE, AXE_SKILL, ONE_HANDED_WEAPON_SKILL, weight, quality, true) {
    init {
        damage = mapOf(
                BLUNT to quality * 0.25,
                CUT to 0.0,
                SLASH to quality * 0.75,
                PIERCING to 0.0,
                CRUSH to 0.0)
    }
}
